### DISCUSSION TITLE
Binding Methods from files inside another toplevel  folder

### DISCUSSION THREAD

**Discussion by craftogrammer** [wailsapp/wails#1360] [Q&A]
Hi

Here's my folder structure. (I am trying Wails V2 beta first time)

![image](https://user-images.githubusercontent.com/52219366/165823546-0650888b-75a7-43a5-97b4-6c2b55156022.png)

My question is how can I avail the methods available in `backend/jpeg/jpeg.go` in javascript?

I am not able to understand how to bind this in `main.go` in root folder

![image](https://user-images.githubusercontent.com/52219366/165823971-9a5f3363-98a8-4e66-aa7a-0092662885f8.png)

Thanks

**Comment by leaanthony** [comment]
Binding in Wails is similar to the concept of controllers in webapps. You bind a struct and all public methods get bound on the JS side. If you want to expose methods on an external package, just consider what exactly you are wanting to pass back and forth to the UI and create your controller based on that. There's good support for passing structs back and forth between Go and JS, but you need to ensure the struct fields have JSON tags, so simply proxying the library methods may not work as you expect. So consider what you want your "API" to look like and try to keep most of your application logic in Go. Good luck!
